### PR TITLE
RR-30 add generic object support

### PR DESCRIPTION
**Pull request by rbrott**
@kellyrm not sure how much of the dashboard code you've examined... and yay JavaScript 😄

### CODE REVIEW

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/RobotDashboard.java [hunk 1]
```diff
@@ -69,9 +70,7 @@ public boolean shouldProcessClass(String className) {
             public void processClass(Class clazz) {
                 if (clazz.isAnnotationPresent(Config.class)) {
                 	Log.i(TAG, String.format("Found config class %s", clazz.getCanonicalName()));
-                    Config annotation = (Config) clazz.getAnnotation(Config.class);
-                    String name = annotation.value().equals("") ? clazz.getSimpleName() : annotation.value();
-                    optionGroups.add(new OptionGroup(clazz, name, prefs));
```
Comment (kellyrm) [inline RobotLib/src/main/java/com/acmerobotics/library/dashboard/RobotDashboard.java:74]: what are going on with these variable names?

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Configuration.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+public class Configuration {
+    private Map<String, Option> options;
+
+    public Configuration() {
+        options = new HashMap<>();
+    }
+
+    public void addOptionsFromClass(Class<?> klass) {
```
Comment (kellyrm) [inline RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Configuration.java:20]: is there a distinction between klass and clazz?

File: dashboard/yarn.lock [hunk 3]
```diff
@@ -910,9 +910,9 @@ babel-preset-jest@^20.0.3:
   dependencies:
```
Comment (kellyrm) [inline dashboard/yarn.lock:910]: that's a lot of dependencies

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
+package com.acmerobotics.library.dashboard.config;
```
Comment (kellyrm) [inline RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java:1]: what is the purpose of putting all this logic in here and using an enum? why not put in the subclasses
could you use this.class? idk if classes work that way

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    public Option(OptionType type) {
+        this.type = type;
+    }
+
+    public OptionType getType() {
+        return type;
+    }
+
+    public JsonElement getJson() {
+        if (type == OptionType.CUSTOM) {
```
Comment (kellyrm) [inline RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java:32]: this seems very un-oop

**Review by kellyrm — APPROVED** [review]
nifty

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/RobotDashboard.java (hunk 1 above)
Comment (rbrott) [reply RobotLib/src/main/java/com/acmerobotics/library/dashboard/RobotDashboard.java:74]: Yeah it's weird... I'll change them all to `klass`

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Configuration.java (hunk 2 above)
Comment (rbrott) [reply RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Configuration.java:20]: Nope 😄

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java (hunk 4 above)
Comment (rbrott) [reply RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java:1]: Yeah I'm going to restructure it better. I don't think `this.class` works that way.

File: RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java (hunk 5 above)
Comment (rbrott) [reply RobotLib/src/main/java/com/acmerobotics/library/dashboard/config/Option.java:32]: Yeah. See above.

File: dashboard/yarn.lock (hunk 3 above)
Comment (rbrott) [reply dashboard/yarn.lock:910]: Yeah. The real top-level dependencies are stored in `dashboard/package.json.` Yarn generates a complete tree including all subdependencies with version numbers for consistency. I don't directly modify this file; it's auto-generated. They recommend to check it into version control.

**Pull request merged**